### ISSUE TITLE
Duplicate extension rules in naming config silently break Naming and Disable Comments checks

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#122]
## Bug Description

When `check.toml` contains two naming rules with the same file extension but different case requirements, both the **Naming** and **Disable Comments** checks silently fail. They report "passed" even when there are files that should be flagged as violations.

## Severity

**Critical** - This can allow naming violations and disable comments to slip through CI checks undetected.

## Steps to Reproduce

1. Create a `check.toml` with duplicate extension rules:

```toml
[code.naming]
enabled = true

[[code.naming.rules]]
extensions = ["ts", "tsx"]
file_case = "kebab-case"
folder_case = "kebab-case"

[[code.naming.rules]]
extensions = ["ts"]  # Duplicate extension with conflicting case
file_case = "snake_case"
folder_case = "snake_case"

[[code.naming.rules]]
extensions = ["py"]
file_case = "snake_case"
folder_case = "snake_case"

[code.quality.disable-comments]
enabled = true
extensions = ["ts", "tsx", "js", "jsx", "py"]
```

2. Run `cm validate config` - it passes validation ✓
3. Run `cm code check` - Naming shows "passed" even with files that violate naming rules
4. Disable Comments also shows "passed" even with disable comments in files

## Expected Behavior

One of the following:
- Configuration validation should reject duplicate extensions across rules, OR
- The tool should use a deterministic rule (first match, last match, etc.) and still find violations, OR
- At minimum, the tool should not silently fail - it should either work correctly or produce an error

## Actual Behavior

- Configuration validation passes
- Naming check reports "passed" (0 violations)
- Disable Comments check reports "passed" (0 violations)
- Files that clearly violate naming conventions are not detected

## Evidence

**With standard check.toml (no duplicates):**
```
✗ Naming: 12 violation(s)
✗ Disable Comments: 32 violation(s)
```

**With duplicate extension config:**
```
✓ Naming: passed
✓ Disable Comments: passed
```

## Environment

- check-my-toolkit version: 1.3.0
- OS: macOS

## Suggested Fix

Either:
1. Add validation to reject configs with duplicate extensions across naming rules
2. Fix the internal logic to handle duplicate extensions deterministically
3. At minimum, log a warning when duplicate extensions are detected

**Comment by chrismlittle123** [comment]
Fixed in commit 393a23c - Extensions are now deduplicated in buildGlobPattern() to prevent glob pattern issues.
